### PR TITLE
add slingshot and handle upload

### PR DESCRIPTION
**Pull request by mixcas**
Need to `meteor npm install` and settings.json is int he sync in the mup folder

### CODE REVIEW

File: app/imports/startup/slingshot.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { Meteor } from 'meteor/meteor';
+import { Random } from 'meteor/random';
+
+Slingshot.fileRestrictions('imageUpload', {
+  allowedFileTypes: ['image/png', 'image/jpeg', 'image/gif'],
+  maxSize: parseInt(Meteor.settings.public.maxUploadSize),
+});
+
+// Misc
+function createFilename(filename) {
```
Comment (varffs) [inline app/imports/startup/slingshot.js]: why parse the variable if not used?

File: app/imports/startup/slingshot.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      if (!this.userId) {
+        throw new Meteor.Error('not-signed-in', 'You must register a user first before uploading a file.');
+      }
+      */
+
+      return true;
+    },
+
+    key(file) {
+      // Store file into an image directory for the user's username.
```
Comment (varffs) [inline app/imports/startup/slingshot.js]: old code comment

File: app/imports/startup/slingshot.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      return 'images/' + createFilename(file.name);
```
Comment (varffs) [inline app/imports/startup/slingshot.js]: maybe we should split uploads a bit more. into year and month? or maybe by location??

File: app/imports/startup/slingshot.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+/* globals Slingshot Random */
+import { Meteor } from 'meteor/meteor';
+import { Random } from 'meteor/random';
+
+Slingshot.fileRestrictions('imageUpload', {
+  allowedFileTypes: ['image/png', 'image/jpeg', 'image/gif'],
```
Comment (varffs) [inline app/imports/startup/slingshot.js]: do we not know exactly what file type we are working with?

File: app/imports/components/posts/NewPost.jsx [hunk 5]
```diff
@@ -16,30 +16,74 @@ export class NewPost extends Component {
...
+
+  handleUpload(error, url) {
     const caption = this.state.caption;
     const fingerprint = this.state.fingerprint;
 
-    addPost.call({
-      photo,
-      caption,
-      fingerprint,
+    this.setState({
```
Comment (varffs) [inline app/imports/components/posts/NewPost.jsx]: perhaps this should only reset on fail as we don't want any more uploads if successful

File: app/imports/components/posts/NewPost.jsx [hunk 6]
```diff
@@ -16,30 +16,74 @@ export class NewPost extends Component {
...
+      addPost.call({
+        caption,
+        photo: url,
+        fingerprint,
+      }, (err, res) => {
+        if (err) {
+          console.log(err);
+        }
+
+        FlowRouter.go('home');
```
Comment (varffs) [inline app/imports/components/posts/NewPost.jsx:63]: this is in the error callback and probably should be outside of it?

**Review by varffs — CHANGES_REQUESTED** [review]
Functions totally fine but some comments that probably need to be addressed

File: app/imports/startup/slingshot.js (hunk 4 above)
Comment (mixcas) [reply app/imports/startup/slingshot.js]: i rly dont know, specially since iOS could open the gallery. but this types are right no?

Comment (varffs) [reply app/imports/startup/slingshot.js]: but as we don't want anything other than a recently taken photo why not ban the other formats?

File: app/imports/startup/slingshot.js (hunk 3 above)
Comment (mixcas) [reply app/imports/startup/slingshot.js]: maybe… later :)

Comment (varffs) [reply app/imports/startup/slingshot.js]: that smells horrible

**Comment by mixcas** [comment]
@varffs there u go

**Pull request merged**